c++/src/kj/compat/http.c++

```diff
@@ -2362,23 +2362,22 @@ public:
   }
 
   Promise<void> write(const void* buffer, size_t size) override {
-    if (size == 0) return kj::READY_NOW;  // can't encode zero-size chunk since it indicates EOF.
+    if (size == 0) co_return;  // can't encode zero-size chunk since it indicates EOF.
 
     auto header = kj::str(kj::hex(size), "\r\n");
     auto parts = kj::heapArray<ArrayPtr<const byte>>(3);
     parts[0] = header.asBytes();
     parts[1] = kj::arrayPtr(reinterpret_cast<const byte*>(buffer), size);
     parts[2] = kj::StringPtr("\r\n").asBytes();
 
-    auto promise = getInner().writeBodyData(parts.asPtr());
-    return promise.attach(kj::mv(header), kj::mv(parts));
+    co_await getInner().writeBodyData(parts.asPtr());
   }
 
   Promise<void> write(ArrayPtr<const ArrayPtr<const byte>> pieces) override {
     uint64_t size = 0;
     for (auto& piece: pieces) size += piece.size();
 
-    if (size == 0) return kj::READY_NOW;  // can't encode zero-size chunk since it indicates EOF.
+    if (size == 0) co_return;  // can't encode zero-size chunk since it indicates EOF.
 
     auto header = kj::str(kj::hex(size), "\r\n");
     auto partsBuilder = kj::heapArrayBuilder<ArrayPtr<const byte>>(pieces.size() + 2);
```

Comment:
Maybe KJ_STACK_ARRAY?

Reply:
That works, but only because our Clang implementation of KJ_STACK_ARRAY doesn't use VLAs. If we were to compile it on GCC, it would try to use a VLA, which aren't allowed in coroutines.

Perhaps we could have a KJ_STACK_ARRAY_NO_VLA?

Reply:
I pushed a version using KJ_STACK_ARRAY in the meantime.

Reply:
Hmm that does seem like a problem. I suspect no compiler is likely ever to support VLAs in coroutines.

Maybe we should just change KJ_STACK_ARRAY to not use VLAs ever, given we only really care about clang which doesn't support them?

Reply:
Yeah, I think that change is pretty reasonable. It'd be nice to be able to safely use KJ_STACK_ARRAY guilt-free in coroutines. I'll see if I can make it.

Reply:
Dropping VLA support from KJ_STACK_ARRAY in #1762.